code/obj/submachine/weapon_vendor.dm

```diff
@@ -21,15 +21,78 @@
 	density = 1
 	opacity = 0
 	anchored = 1
+	flags = TGUI_INTERACTIVE
 
 	var/sound_token = 'sound/machines/capsulebuy.ogg'
 	var/sound_buy = 'sound/machines/spend.ogg'
-	var/temp = null
 	var/list/credits = list("Sidearm" = 0, "Loadout" = 0, "Utility" = 0, "Assistant" = 0)
 	var/list/datum/materiel_stock = list()
 	var/token_accepted = /obj/item/requisition_token
 	var/log_purchase = FALSE
 
+	ui_interact(mob/user, datum/tgui/ui)
+		ui = tgui_process.try_update_ui(user, src, ui)
+		if(!ui)
+			ui = new(user, src, "WeaponVendor", src.name)
+			ui.open()
+
+	ui_static_data(mob/user)
+		. = list("stock" = list())
+
+		for (var/datum/materiel/M in materiel_stock)
```

Comment:
candidate for `as anything`

Reply:
Could you expand on this? Not really sure what you mean. Are you talking about the `as()` part i keep seeing in for loops?